File: 1528.shuffle-string.js
/*
 * @lc app=leetcode id=1528 lang=javascript
 *
 * [1528] Shuffle String
 */

// @lc code=start
/**
 * @param {string} s
 * @param {number[]} indices
 * @return {string}
 */
var restoreString = function(s, indices) {
    const chars = [];
    for (let i = 0; i < s.length; i++) {
        chars[indices[i]] = s[i];
    }
    return chars.join('');
};
// @lc code=end

File: 1528.shuffle-string.py
#
# @lc app=leetcode id=1528 lang=python
#
# [1528] Shuffle String
#

# @lc code=start
class Solution(object):
    def restoreString(self, s, indices):
        """
        :type s: str
        :type indices: List[int]
        :rtype: str
        """

        result = [None] * len(s)
        for i, index in enumerate(indices):
            result[index] = s[i]

        return "".join(result)
# @lc code=end
